### DISCUSSION TITLE
Newbie question..how to create container from a strictly local gcc executable?

### DISCUSSION THREAD

**Discussion by k9cfmark** [containers/podman#21374] [Q&A]
Very newbie question.
I can pull and execute the hello world example so the install on my PI is working.
How do I create a new locally written gcc program that doesn’t exist in a GitHub or docker repository?
Can someone point me to a dummies guide that gets me past THIS threshold?

**Comment by afbjorklund** [comment]
The easiest is using a Makefile and a Containerfile, but if it is a single binary you can also just "import" it.

```C
#include <stdio.h>
int main() { puts("Hello, World!"); }
```

```console
$ gcc -o hello -static hello.c
$ tar cf hello.tar hello
$ podman import hello.tar hello
$ podman run --rm hello /hello
Hello, World!
```

https://github.com/containers/PodmanHello

**Reply by k9cfmark** [reply]
Thank you!!  
it seemed if you were already using repositories and had committed your base line, the instructions might make sense…..but I haven’t been, so it didn’t.
I’ll try the TAR and Import path. 
thank so much!

**Reply by afbjorklund** [reply]
The repository was supposed to have the "long" answer, including the Containerfile.

You can open new issues there ("PodmanHello"), if something is unclear about it?

**Comment by rhatdan** [comment]
You can also do a Containerfile with the following
from scratch
add hello /hello
podman build 
podman run
